assets/src/edit-story/app/media/utils/getAttachmentFromResource.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const getAttachmentFromResource = ({
+  id,
+  posterId,
+  poster,
+  src,
+  oWidth,
```

Comment:
Made a note above. Really close, but some nuances. I don't think `getAttachmentFromResource` is called correctly in the `apiProvider.js`. The delta is possibly very small. For instance, `oWidth` does not exist in the `Resource` type. I think instead it's `width`. There might be a few more discrepancies like this.

Reply:
Cool! So are you proposing to return `width` (height, etc.) here instead of use `oWidth` and specific data from API directly in order to normalize it? Same for File Resource object?

Reply:
The method is called `getAttachmentFromResource()` so I assumed the the input is the `Resource` type and it returns `Attachment`. E.g. something like this:

```
getAttachmentFromResource(resource: Resource): Attachment
```

But that's not the case in your code. The input `Resource` type does not have `oWidth` field. Although the output `Attachment` does. I'm not sure how many other mismatches there are in these two types. But mixing them would be an obvious source of bugs. Unfortunately the typing setup makes it unenforceable in our code, so we have to be careful.